jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/SparkParallelismTracker.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ Copyright (c) 2014 by Contributors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+package ml.dmlc.xgboost4j.scala.spark
+
+import ml.dmlc.xgboost4j.java.XGBoostError
+import org.apache.spark.SparkContext
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Future}
+import scala.concurrent.ExecutionContext.Implicits.global
+
+/**
+  * A tracker that periodically checks the number of alive Spark executor cores.
+  * Throws an exception when the number of alive cores is less than nWorkers.
+  *
+  * @param sc The SparkContext object
+  * @param checkInterval The interval to check executor status in milliseconds.
+  * @param nWorkers nWorkers used in an XGBoost Job
+  */
+private[spark] class SparkParallelismTracker(
+    sc: SparkContext,
+    checkInterval: Long,
+    nWorkers: Int) {
+  private[this] var isRunning = true
+  private[this] val trackerThread = Thread.currentThread()
+
+  private[this] def check(): Unit = {
+    val currentNumExecutors = sc.getExecutorMemoryStatus.size - 1
+    val numCoresPerExecutor = sc.getConf.getInt("spark.executor.cores", 1)
```

Comment:
With standalone (maybe also Mesos), users can skip setting this parameter and RM will assign all available cores in `Worker` to an executor. In this case, this info will be inaccurate

Reply:
if you cannot find any other way to get cores info, you may want to call REST API to fetch the info

Reply:
Cool. Done~~